posthog/api/capture.py

```diff
@@ -451,11 +470,9 @@ def is_randomly_partitioned(candidate_partition_key: str) -> bool:
         Whether the given partition key should be used.
     """
     if settings.PARTITION_KEY_AUTOMATIC_OVERRIDE_ENABLED:
-
         has_capacity = LIMITER.consume(candidate_partition_key)
 
-        if has_capacity is False:
-
+        if has_capacity is False or candidate_partition_key.rsplit(":")[1] in LIKELY_ANONYMOUS_IDS:
```

Comment:
To avoid calling rsplit on the hot path, we can move that check to `capture_internal`, with a `if distinct_id.lower() not in LIKELY_ANONYMOUS_IDS:` check around the `candidate_partition_key` logic.